stl/inc/memory

```diff
@@ -1578,20 +1762,211 @@ private:
+    _NODISCARD shared_ptr<_Ty> make_shared(_Types&&... _Args) { // make a shared_ptr to non-array object
     const auto _Rx = new _Ref_count_obj2<_Ty>(_STD forward<_Types>(_Args)...);
     shared_ptr<_Ty> _Ret;
     _Ret._Set_ptr_rep_and_enable_shared(_STD addressof(_Rx->_Storage._Value), _Rx);
     return _Ret;
 }
 
+#if _HAS_CXX20 && __cpp_concepts
+template <class _Ty>
+requires _STD is_unbounded_array_v<_Ty> _NODISCARD shared_ptr<_Ty> make_shared(
+    size_t _Count) { // make a shared_ptr to an unbounded array
+    using _Refc = _Ref_count_unbounded_array<_Ty>;
+    using _Storage =
+        _STD aligned_storage_t<(_STD max)(alignof(_Ty), alignof(_Refc)), (_STD max)(alignof(_Ty), alignof(_Refc))>;
+    constexpr size_t _Rx_size_div_by_storage_size = (_STD max)(sizeof(_Refc) / sizeof(_Storage), size_t(1));
+    constexpr double _Ty_size_div_by_storage_size =
+        static_cast<double>(sizeof(_STD remove_extent_t<_Ty>)) / static_cast<double>(sizeof(_Storage));
+    const auto _Rx = reinterpret_cast<_Refc*>(
+        new _Storage[static_cast<size_t>(ceil(_Count * _Ty_size_div_by_storage_size)) + _Rx_size_div_by_storage_size]);
+    new (_Rx) _Refc(_Count);
+    shared_ptr<_Ty> _Ret;
+    _Ret._Set_ptr_rep_and_enable_shared(_Rx->_Get_ptr(), _Rx);
+    return _Ret;
+}
+
+template <class _Ty>
+requires _STD is_unbounded_array_v<_Ty> _NODISCARD shared_ptr<_Ty> make_shared(size_t _Count,
+    const _STD remove_extent_t<_Ty>& _Val) { // make a shared_ptr to an unbounded array
+    using _Refc = _Ref_count_unbounded_array<_Ty>;
+    using _Storage =
+        _STD aligned_storage_t<(_STD max)(alignof(_Ty), alignof(_Refc)), (_STD max)(alignof(_Ty), alignof(_Refc))>;
+    constexpr size_t _Rx_size_div_by_storage_size = (_STD max)(sizeof(_Refc) / sizeof(_Storage), size_t(1));
+    constexpr double _Ty_size_div_by_storage_size =
+        static_cast<double>(sizeof(_STD remove_extent_t<_Ty>)) / static_cast<double>(sizeof(_Storage));
+    const auto _Rx = reinterpret_cast<_Refc*>(
+        new _Storage[static_cast<size_t>(ceil(_Count * _Ty_size_div_by_storage_size)) + _Rx_size_div_by_storage_size]);
+    new (_Rx) _Refc(_Count, _Val);
+    shared_ptr<_Ty> _Ret;
+    _Ret._Set_ptr_rep_and_enable_shared(_Rx->_Get_ptr(), _Rx);
+    return _Ret;
+}
+
+template <class _Ty>
+requires _STD is_bounded_array_v<_Ty> _NODISCARD shared_ptr<_Ty> make_shared() { // make a shared_ptr to a bounded array
+    const auto _Rx = new _Ref_count_bounded_array<_Ty>();
+    shared_ptr<_Ty> _Ret;
+    _Ret._Set_ptr_rep_and_enable_shared(_Rx->_Storage._Value, _Rx);
+    return _Ret;
+}
+
+template <class _Ty>
+requires _STD is_bounded_array_v<_Ty> _NODISCARD shared_ptr<_Ty> make_shared(
+    const _STD remove_extent_t<_Ty>& _Val) { // make a shared_ptr to a bounded array
+    const auto _Rx = new _Ref_count_bounded_array<_Ty>(_Val);
+    shared_ptr<_Ty> _Ret;
+    _Ret._Set_ptr_rep_and_enable_shared(_Rx->_Storage._Value, _Rx);
+    return _Ret;
+}
+#endif // _HAS_CXX20

```

Comment:
There should be a newline after this `#endif`.

Reply:
Fixed.